File: api-registro-escolar/src/main/java/com/dt/registroescolar/api_registro_escolar/dto/ProfesorDTO.java
package com.dt.registroescolar.api_registro_escolar.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.PastOrPresent;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ProfesorDTO extends PersonaDTO {
    @JsonProperty("id") // Esto dice: "usa 'id' en el JSON"
    private Long idPersona;
//    private Long idPersona;
//    private String nombre;
//    private String apellido;
//    private String email;
//    private String telefono;
//    private LocalDate fechaNacimiento;
    @NotBlank(message = "La especialidad es obligatoria")
    @Size(max = 100, message = "La especialidad no debe exceder los 100 caracteres")
    private String especialidad;

    @PastOrPresent(message = "La fecha de contratación no puede ser futura")
    private LocalDate fechaContratacion;
}

File: registro-escolar-frontend/src/app/models/profesor.dto.ts
// export interface ProfesorDTO {
//   idPersona?: number;
//   nombre: string;
//   apellido: string;
//   email: string;
//   telefono: string;
//   fechaNacimiento: string;
//   especialidad: string;
// }

import { PersonaDTO } from './persona.dto';

export interface ProfesorDTO extends PersonaDTO {
  // idPersona?: number; // Este puede ser redundante si ya usas 'id' de PersonaDTO
  especialidad: string;
  fechaContratacion: string; // formato 'YYYY-MM-DD'
}